### PR TITLE
Handle spec level timeouts

### PR DESCRIPTION
**Pull request by roshan04**
Changes for adding spec level timeouts.

### CODE REVIEW

File: bin/helpers/constants.js [hunk 1]
```diff
@@ -62,7 +62,9 @@ const userMessages = {
...
+  BUILD_FAILED_ERROR: "The above stacktrace has been thrown by Cypress when we tried to run your build. If your test suite requires npm dependencies then please specify them on browserstack.json. Read more at " +  chalk.blueBright("https://www.browserstack.com/docs/automate/cypress/npm-packages") + ". Also, we recommend you to try running the build locally using ‘cypress run’ and if it works fine then please reach out to support at " + chalk.blueBright("https://www.browserstack.com/contact#technical-support"),
+  SPEC_LIMIT_WARNING: "Value for the 'spec_timeout' key not in the 1-120 range. Going ahead with 30 mins as the default spec timeout. Read more about how to specify the option in https://browserstack.com/docs/automate/cypress/spec-timeout ",
```
Comment (nagpalkaran95) [inline bin/helpers/constants.js:66]:
```suggestion
  SPEC_LIMIT_WARNING: "Value for the 'spec_timeout' key not in the 1-120 range. Going ahead with 30 mins as the default spec timeout. Read more about how to specify the option in https://browserstack.com/docs/automate/cypress/spec-timeout",
```

File: bin/helpers/utils.js [hunk 2]
```diff
@@ -389,10 +406,26 @@ exports.fixCommaSeparatedString = (string) => {
...
+  }
+
+  return false;
+}
+
 exports.isTrueString = value => (!this.isUndefined(value) && value.toString().toLowerCase() === 'true');
 
 exports.isFloat = (value) => Number(value) && Number(value) % 1 !== 0;
 
+exports.isInteger = (value) => Number.isInteger(value);
```
Comment (nagpalkaran95) [inline bin/helpers/utils.js:427]: add tests for these too

File: bin/helpers/utils.js [hunk 3]
```diff
@@ -389,10 +406,26 @@ exports.fixCommaSeparatedString = (string) => {
 
 exports.isUndefined = value => (value === undefined || value === null);
 
+exports.isPositiveInteger = (str) => {
```
Comment (nagpalkaran95) [inline bin/helpers/utils.js:409]: add tests for these too

File: bin/runner.js [hunk 4]
```diff
@@ -168,6 +168,12 @@ var argv = yargs
           type: "string",
           default: undefined
         },
+        't': {
```
Comment (nagpalkaran95) [inline bin/runner.js:171]: Need to add help section for this arg. Also specify `mins` when giving range in messgae

**Review by nagpalkaran95 — CHANGES_REQUESTED** [review]

File: bin/runner.js (hunk 4 above)
Comment (roshan04) [reply bin/runner.js:171]: by help section do you mean the description have added it under `SPEC_TIMEOUT`

File: bin/helpers/utils.js (hunk 3 above)
Comment (roshan04) [reply bin/helpers/utils.js:409]: done

File: bin/helpers/utils.js (hunk 2 above)
Comment (roshan04) [reply bin/helpers/utils.js:427]: done

File: bin/helpers/constants.js (hunk 1 above)
Comment (roshan04) [reply bin/helpers/constants.js]: done

**Review by nagpalkaran95 — DISMISSED** [review]

**Review by nagpalkaran95 — APPROVED** [review]

**Pull request merged**